packages/@lwc/integration-karma/test/custom-elements/index.spec.js

```diff
@@ -208,6 +208,34 @@ if (SUPPORTS_CUSTOM_ELEMENTS) {
                 });
         });
 
+        it('vanilla element should be an instance of the right class', () => {
+            const tagName = 'x-check-ctor-instanceof';
+            class MyElement extends HTMLElement {}
+
+            const whenDefinedPromiseBeforeDefine = customElements.whenDefined(tagName);
+
+            customElements.define(tagName, MyElement);
+            const elm = document.createElement(tagName);
+
+            // check element prototype
+            expect(elm instanceof MyElement).toEqual(true);
+            expect(elm.constructor).toBe(MyElement);
+
+            // check cE.get()
+            const Ctor = customElements.get(tagName);
+            expect(Ctor).toBe(MyElement);
+
+            // check cE.whenDefined()
+            return whenDefinedPromiseBeforeDefine
+                .then((Ctor) => {
+                    expect(Ctor).toBe(MyElement);
+                    return customElements.whenDefined(tagName);
```

Comment:
Is this line required? The then block is only called when `customElements.whenDefined(tagName)` promise has been resolved.

Reply:
Yes I'm checking two things – `whenDefined()` when called before `define()`, and `whenDefined` called after. This goes through two code paths. I'll add a comment